### PR TITLE
fix how we retarget flutter actions

### PR DESCRIPTION
**Pull request by devoncarew**
This addresses an issue I saw this weekend. I had two project windows open against two separate flutter projects. I hit run in one project; when the app came up, the reload icon became enabled both in the current project, but also in the 2nd open project window. Hitting reload in either window would reload the app running for the first window.

- switch from using `ActionManager` to track globally registered actions, to using a new tracker that's scoped to projects
- additionally, fix an issue in reload on save where we didn't reload when the focused dart editor wasn't in the module for the app currently running. From dogfooding, it's surprising when the app doesn't reload (users get a sense that cmd-s means reload).

@pq

### CODE REVIEW

File: src/io/flutter/run/FlutterReloadManager.java [hunk 1]
```diff
@@ -327,14 +319,14 @@ private boolean hasErrors(@NotNull Project project, @NotNull Module module, @Not
     }
   }
 
-  private static boolean shouldBlockReload(@NotNull DartServerData.DartError error, @NotNull Module module) {
+  private static boolean shouldBlockReload(@NotNull DartServerData.DartError error, @NonNull Project project, @Nullable Module module) {
```
Comment (pq) [inline src/io/flutter/run/FlutterReloadManager.java]: I'm assuming `NonNull` was a typo and you mean `NotNull`?

File: src/io/flutter/run/FlutterReloadManager.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
  */
 package io.flutter.run;
 
+import com.android.annotations.NonNull;
```
Comment (pq) [inline src/io/flutter/run/FlutterReloadManager.java]: See below: I think we don't want this import?

Comment (devoncarew) [reply src/io/flutter/run/FlutterReloadManager.java]: oops - removing

File: src/io/flutter/run/FlutterReloadManager.java (hunk 1 above)
Comment (devoncarew) [reply src/io/flutter/run/FlutterReloadManager.java]: Good catch! done

**Pull request merged**